{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ba1c2047",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-23 10:59:02.171779: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory\n",
      "2021-12-23 10:59:02.171803: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from keras.models import Sequential\n",
    "from keras.layers import Dense\n",
    "from keras.wrappers.scikit_learn import KerasClassifier\n",
    "from keras.utils import np_utils\n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.pipeline import Pipeline\n",
    "\n",
    "from sklearn.datasets import load_iris\n",
    "\n",
    "#from prova1.config.AppConfig import DATAPATH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a8fd7c1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sepal.length</td>\n",
       "      <td>sepal.width</td>\n",
       "      <td>petal.length</td>\n",
       "      <td>petal.width</td>\n",
       "      <td>variety</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.1</td>\n",
       "      <td>3.5</td>\n",
       "      <td>1.4</td>\n",
       "      <td>.2</td>\n",
       "      <td>Setosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4.9</td>\n",
       "      <td>3</td>\n",
       "      <td>1.4</td>\n",
       "      <td>.2</td>\n",
       "      <td>Setosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>1.3</td>\n",
       "      <td>.2</td>\n",
       "      <td>Setosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4.6</td>\n",
       "      <td>3.1</td>\n",
       "      <td>1.5</td>\n",
       "      <td>.2</td>\n",
       "      <td>Setosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>6.7</td>\n",
       "      <td>3</td>\n",
       "      <td>5.2</td>\n",
       "      <td>2.3</td>\n",
       "      <td>Virginica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>6.3</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5</td>\n",
       "      <td>1.9</td>\n",
       "      <td>Virginica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>6.5</td>\n",
       "      <td>3</td>\n",
       "      <td>5.2</td>\n",
       "      <td>2</td>\n",
       "      <td>Virginica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>6.2</td>\n",
       "      <td>3.4</td>\n",
       "      <td>5.4</td>\n",
       "      <td>2.3</td>\n",
       "      <td>Virginica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>5.9</td>\n",
       "      <td>3</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.8</td>\n",
       "      <td>Virginica</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                0            1             2            3          4\n",
       "0    sepal.length  sepal.width  petal.length  petal.width    variety\n",
       "1             5.1          3.5           1.4           .2     Setosa\n",
       "2             4.9            3           1.4           .2     Setosa\n",
       "3             4.7          3.2           1.3           .2     Setosa\n",
       "4             4.6          3.1           1.5           .2     Setosa\n",
       "..            ...          ...           ...          ...        ...\n",
       "146           6.7            3           5.2          2.3  Virginica\n",
       "147           6.3          2.5             5          1.9  Virginica\n",
       "148           6.5            3           5.2            2  Virginica\n",
       "149           6.2          3.4           5.4          2.3  Virginica\n",
       "150           5.9            3           5.1          1.8  Virginica\n",
       "\n",
       "[151 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframe = pd.read_csv(\"iris.csv\", header=None)\n",
    "dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1d606dc7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['sepal.length', 'sepal.width', 'petal.length', 'petal.width',\n",
       "        'variety'],\n",
       "       ['5.1', '3.5', '1.4', '.2', 'Setosa'],\n",
       "       ['4.9', '3', '1.4', '.2', 'Setosa'],\n",
       "       ['4.7', '3.2', '1.3', '.2', 'Setosa'],\n",
       "       ['4.6', '3.1', '1.5', '.2', 'Setosa'],\n",
       "       ['5', '3.6', '1.4', '.2', 'Setosa'],\n",
       "       ['5.4', '3.9', '1.7', '.4', 'Setosa'],\n",
       "       ['4.6', '3.4', '1.4', '.3', 'Setosa'],\n",
       "       ['5', '3.4', '1.5', '.2', 'Setosa'],\n",
       "       ['4.4', '2.9', '1.4', '.2', 'Setosa'],\n",
       "       ['4.9', '3.1', '1.5', '.1', 'Setosa'],\n",
       "       ['5.4', '3.7', '1.5', '.2', 'Setosa'],\n",
       "       ['4.8', '3.4', '1.6', '.2', 'Setosa'],\n",
       "       ['4.8', '3', '1.4', '.1', 'Setosa'],\n",
       "       ['4.3', '3', '1.1', '.1', 'Setosa'],\n",
       "       ['5.8', '4', '1.2', '.2', 'Setosa'],\n",
       "       ['5.7', '4.4', '1.5', '.4', 'Setosa'],\n",
       "       ['5.4', '3.9', '1.3', '.4', 'Setosa'],\n",
       "       ['5.1', '3.5', '1.4', '.3', 'Setosa'],\n",
       "       ['5.7', '3.8', '1.7', '.3', 'Setosa'],\n",
       "       ['5.1', '3.8', '1.5', '.3', 'Setosa'],\n",
       "       ['5.4', '3.4', '1.7', '.2', 'Setosa'],\n",
       "       ['5.1', '3.7', '1.5', '.4', 'Setosa'],\n",
       "       ['4.6', '3.6', '1', '.2', 'Setosa'],\n",
       "       ['5.1', '3.3', '1.7', '.5', 'Setosa'],\n",
       "       ['4.8', '3.4', '1.9', '.2', 'Setosa'],\n",
       "       ['5', '3', '1.6', '.2', 'Setosa'],\n",
       "       ['5', '3.4', '1.6', '.4', 'Setosa'],\n",
       "       ['5.2', '3.5', '1.5', '.2', 'Setosa'],\n",
       "       ['5.2', '3.4', '1.4', '.2', 'Setosa'],\n",
       "       ['4.7', '3.2', '1.6', '.2', 'Setosa'],\n",
       "       ['4.8', '3.1', '1.6', '.2', 'Setosa'],\n",
       "       ['5.4', '3.4', '1.5', '.4', 'Setosa'],\n",
       "       ['5.2', '4.1', '1.5', '.1', 'Setosa'],\n",
       "       ['5.5', '4.2', '1.4', '.2', 'Setosa'],\n",
       "       ['4.9', '3.1', '1.5', '.2', 'Setosa'],\n",
       "       ['5', '3.2', '1.2', '.2', 'Setosa'],\n",
       "       ['5.5', '3.5', '1.3', '.2', 'Setosa'],\n",
       "       ['4.9', '3.6', '1.4', '.1', 'Setosa'],\n",
       "       ['4.4', '3', '1.3', '.2', 'Setosa'],\n",
       "       ['5.1', '3.4', '1.5', '.2', 'Setosa'],\n",
       "       ['5', '3.5', '1.3', '.3', 'Setosa'],\n",
       "       ['4.5', '2.3', '1.3', '.3', 'Setosa'],\n",
       "       ['4.4', '3.2', '1.3', '.2', 'Setosa'],\n",
       "       ['5', '3.5', '1.6', '.6', 'Setosa'],\n",
       "       ['5.1', '3.8', '1.9', '.4', 'Setosa'],\n",
       "       ['4.8', '3', '1.4', '.3', 'Setosa'],\n",
       "       ['5.1', '3.8', '1.6', '.2', 'Setosa'],\n",
       "       ['4.6', '3.2', '1.4', '.2', 'Setosa'],\n",
       "       ['5.3', '3.7', '1.5', '.2', 'Setosa'],\n",
       "       ['5', '3.3', '1.4', '.2', 'Setosa'],\n",
       "       ['7', '3.2', '4.7', '1.4', 'Versicolor'],\n",
       "       ['6.4', '3.2', '4.5', '1.5', 'Versicolor'],\n",
       "       ['6.9', '3.1', '4.9', '1.5', 'Versicolor'],\n",
       "       ['5.5', '2.3', '4', '1.3', 'Versicolor'],\n",
       "       ['6.5', '2.8', '4.6', '1.5', 'Versicolor'],\n",
       "       ['5.7', '2.8', '4.5', '1.3', 'Versicolor'],\n",
       "       ['6.3', '3.3', '4.7', '1.6', 'Versicolor'],\n",
       "       ['4.9', '2.4', '3.3', '1', 'Versicolor'],\n",
       "       ['6.6', '2.9', '4.6', '1.3', 'Versicolor'],\n",
       "       ['5.2', '2.7', '3.9', '1.4', 'Versicolor'],\n",
       "       ['5', '2', '3.5', '1', 'Versicolor'],\n",
       "       ['5.9', '3', '4.2', '1.5', 'Versicolor'],\n",
       "       ['6', '2.2', '4', '1', 'Versicolor'],\n",
       "       ['6.1', '2.9', '4.7', '1.4', 'Versicolor'],\n",
       "       ['5.6', '2.9', '3.6', '1.3', 'Versicolor'],\n",
       "       ['6.7', '3.1', '4.4', '1.4', 'Versicolor'],\n",
       "       ['5.6', '3', '4.5', '1.5', 'Versicolor'],\n",
       "       ['5.8', '2.7', '4.1', '1', 'Versicolor'],\n",
       "       ['6.2', '2.2', '4.5', '1.5', 'Versicolor'],\n",
       "       ['5.6', '2.5', '3.9', '1.1', 'Versicolor'],\n",
       "       ['5.9', '3.2', '4.8', '1.8', 'Versicolor'],\n",
       "       ['6.1', '2.8', '4', '1.3', 'Versicolor'],\n",
       "       ['6.3', '2.5', '4.9', '1.5', 'Versicolor'],\n",
       "       ['6.1', '2.8', '4.7', '1.2', 'Versicolor'],\n",
       "       ['6.4', '2.9', '4.3', '1.3', 'Versicolor'],\n",
       "       ['6.6', '3', '4.4', '1.4', 'Versicolor'],\n",
       "       ['6.8', '2.8', '4.8', '1.4', 'Versicolor'],\n",
       "       ['6.7', '3', '5', '1.7', 'Versicolor'],\n",
       "       ['6', '2.9', '4.5', '1.5', 'Versicolor'],\n",
       "       ['5.7', '2.6', '3.5', '1', 'Versicolor'],\n",
       "       ['5.5', '2.4', '3.8', '1.1', 'Versicolor'],\n",
       "       ['5.5', '2.4', '3.7', '1', 'Versicolor'],\n",
       "       ['5.8', '2.7', '3.9', '1.2', 'Versicolor'],\n",
       "       ['6', '2.7', '5.1', '1.6', 'Versicolor'],\n",
       "       ['5.4', '3', '4.5', '1.5', 'Versicolor'],\n",
       "       ['6', '3.4', '4.5', '1.6', 'Versicolor'],\n",
       "       ['6.7', '3.1', '4.7', '1.5', 'Versicolor'],\n",
       "       ['6.3', '2.3', '4.4', '1.3', 'Versicolor'],\n",
       "       ['5.6', '3', '4.1', '1.3', 'Versicolor'],\n",
       "       ['5.5', '2.5', '4', '1.3', 'Versicolor'],\n",
       "       ['5.5', '2.6', '4.4', '1.2', 'Versicolor'],\n",
       "       ['6.1', '3', '4.6', '1.4', 'Versicolor'],\n",
       "       ['5.8', '2.6', '4', '1.2', 'Versicolor'],\n",
       "       ['5', '2.3', '3.3', '1', 'Versicolor'],\n",
       "       ['5.6', '2.7', '4.2', '1.3', 'Versicolor'],\n",
       "       ['5.7', '3', '4.2', '1.2', 'Versicolor'],\n",
       "       ['5.7', '2.9', '4.2', '1.3', 'Versicolor'],\n",
       "       ['6.2', '2.9', '4.3', '1.3', 'Versicolor'],\n",
       "       ['5.1', '2.5', '3', '1.1', 'Versicolor'],\n",
       "       ['5.7', '2.8', '4.1', '1.3', 'Versicolor'],\n",
       "       ['6.3', '3.3', '6', '2.5', 'Virginica'],\n",
       "       ['5.8', '2.7', '5.1', '1.9', 'Virginica'],\n",
       "       ['7.1', '3', '5.9', '2.1', 'Virginica'],\n",
       "       ['6.3', '2.9', '5.6', '1.8', 'Virginica'],\n",
       "       ['6.5', '3', '5.8', '2.2', 'Virginica'],\n",
       "       ['7.6', '3', '6.6', '2.1', 'Virginica'],\n",
       "       ['4.9', '2.5', '4.5', '1.7', 'Virginica'],\n",
       "       ['7.3', '2.9', '6.3', '1.8', 'Virginica'],\n",
       "       ['6.7', '2.5', '5.8', '1.8', 'Virginica'],\n",
       "       ['7.2', '3.6', '6.1', '2.5', 'Virginica'],\n",
       "       ['6.5', '3.2', '5.1', '2', 'Virginica'],\n",
       "       ['6.4', '2.7', '5.3', '1.9', 'Virginica'],\n",
       "       ['6.8', '3', '5.5', '2.1', 'Virginica'],\n",
       "       ['5.7', '2.5', '5', '2', 'Virginica'],\n",
       "       ['5.8', '2.8', '5.1', '2.4', 'Virginica'],\n",
       "       ['6.4', '3.2', '5.3', '2.3', 'Virginica'],\n",
       "       ['6.5', '3', '5.5', '1.8', 'Virginica'],\n",
       "       ['7.7', '3.8', '6.7', '2.2', 'Virginica'],\n",
       "       ['7.7', '2.6', '6.9', '2.3', 'Virginica'],\n",
       "       ['6', '2.2', '5', '1.5', 'Virginica'],\n",
       "       ['6.9', '3.2', '5.7', '2.3', 'Virginica'],\n",
       "       ['5.6', '2.8', '4.9', '2', 'Virginica'],\n",
       "       ['7.7', '2.8', '6.7', '2', 'Virginica'],\n",
       "       ['6.3', '2.7', '4.9', '1.8', 'Virginica'],\n",
       "       ['6.7', '3.3', '5.7', '2.1', 'Virginica'],\n",
       "       ['7.2', '3.2', '6', '1.8', 'Virginica'],\n",
       "       ['6.2', '2.8', '4.8', '1.8', 'Virginica'],\n",
       "       ['6.1', '3', '4.9', '1.8', 'Virginica'],\n",
       "       ['6.4', '2.8', '5.6', '2.1', 'Virginica'],\n",
       "       ['7.2', '3', '5.8', '1.6', 'Virginica'],\n",
       "       ['7.4', '2.8', '6.1', '1.9', 'Virginica'],\n",
       "       ['7.9', '3.8', '6.4', '2', 'Virginica'],\n",
       "       ['6.4', '2.8', '5.6', '2.2', 'Virginica'],\n",
       "       ['6.3', '2.8', '5.1', '1.5', 'Virginica'],\n",
       "       ['6.1', '2.6', '5.6', '1.4', 'Virginica'],\n",
       "       ['7.7', '3', '6.1', '2.3', 'Virginica'],\n",
       "       ['6.3', '3.4', '5.6', '2.4', 'Virginica'],\n",
       "       ['6.4', '3.1', '5.5', '1.8', 'Virginica'],\n",
       "       ['6', '3', '4.8', '1.8', 'Virginica'],\n",
       "       ['6.9', '3.1', '5.4', '2.1', 'Virginica'],\n",
       "       ['6.7', '3.1', '5.6', '2.4', 'Virginica'],\n",
       "       ['6.9', '3.1', '5.1', '2.3', 'Virginica'],\n",
       "       ['5.8', '2.7', '5.1', '1.9', 'Virginica'],\n",
       "       ['6.8', '3.2', '5.9', '2.3', 'Virginica'],\n",
       "       ['6.7', '3.3', '5.7', '2.5', 'Virginica'],\n",
       "       ['6.7', '3', '5.2', '2.3', 'Virginica'],\n",
       "       ['6.3', '2.5', '5', '1.9', 'Virginica'],\n",
       "       ['6.5', '3', '5.2', '2', 'Virginica'],\n",
       "       ['6.2', '3.4', '5.4', '2.3', 'Virginica'],\n",
       "       ['5.9', '3', '5.1', '1.8', 'Virginica']], dtype=object)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset = dataframe.values\n",
    "dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6f452a07",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([['5.1', '3.5', '1.4', '.2'],\n",
       "        ['4.9', '3', '1.4', '.2'],\n",
       "        ['4.7', '3.2', '1.3', '.2'],\n",
       "        ['4.6', '3.1', '1.5', '.2'],\n",
       "        ['5', '3.6', '1.4', '.2'],\n",
       "        ['5.4', '3.9', '1.7', '.4'],\n",
       "        ['4.6', '3.4', '1.4', '.3'],\n",
       "        ['5', '3.4', '1.5', '.2'],\n",
       "        ['4.4', '2.9', '1.4', '.2'],\n",
       "        ['4.9', '3.1', '1.5', '.1'],\n",
       "        ['5.4', '3.7', '1.5', '.2'],\n",
       "        ['4.8', '3.4', '1.6', '.2'],\n",
       "        ['4.8', '3', '1.4', '.1'],\n",
       "        ['4.3', '3', '1.1', '.1'],\n",
       "        ['5.8', '4', '1.2', '.2'],\n",
       "        ['5.7', '4.4', '1.5', '.4'],\n",
       "        ['5.4', '3.9', '1.3', '.4'],\n",
       "        ['5.1', '3.5', '1.4', '.3'],\n",
       "        ['5.7', '3.8', '1.7', '.3'],\n",
       "        ['5.1', '3.8', '1.5', '.3'],\n",
       "        ['5.4', '3.4', '1.7', '.2'],\n",
       "        ['5.1', '3.7', '1.5', '.4'],\n",
       "        ['4.6', '3.6', '1', '.2'],\n",
       "        ['5.1', '3.3', '1.7', '.5'],\n",
       "        ['4.8', '3.4', '1.9', '.2'],\n",
       "        ['5', '3', '1.6', '.2'],\n",
       "        ['5', '3.4', '1.6', '.4'],\n",
       "        ['5.2', '3.5', '1.5', '.2'],\n",
       "        ['5.2', '3.4', '1.4', '.2'],\n",
       "        ['4.7', '3.2', '1.6', '.2'],\n",
       "        ['4.8', '3.1', '1.6', '.2'],\n",
       "        ['5.4', '3.4', '1.5', '.4'],\n",
       "        ['5.2', '4.1', '1.5', '.1'],\n",
       "        ['5.5', '4.2', '1.4', '.2'],\n",
       "        ['4.9', '3.1', '1.5', '.2'],\n",
       "        ['5', '3.2', '1.2', '.2'],\n",
       "        ['5.5', '3.5', '1.3', '.2'],\n",
       "        ['4.9', '3.6', '1.4', '.1'],\n",
       "        ['4.4', '3', '1.3', '.2'],\n",
       "        ['5.1', '3.4', '1.5', '.2'],\n",
       "        ['5', '3.5', '1.3', '.3'],\n",
       "        ['4.5', '2.3', '1.3', '.3'],\n",
       "        ['4.4', '3.2', '1.3', '.2'],\n",
       "        ['5', '3.5', '1.6', '.6'],\n",
       "        ['5.1', '3.8', '1.9', '.4'],\n",
       "        ['4.8', '3', '1.4', '.3'],\n",
       "        ['5.1', '3.8', '1.6', '.2'],\n",
       "        ['4.6', '3.2', '1.4', '.2'],\n",
       "        ['5.3', '3.7', '1.5', '.2'],\n",
       "        ['5', '3.3', '1.4', '.2'],\n",
       "        ['7', '3.2', '4.7', '1.4'],\n",
       "        ['6.4', '3.2', '4.5', '1.5'],\n",
       "        ['6.9', '3.1', '4.9', '1.5'],\n",
       "        ['5.5', '2.3', '4', '1.3'],\n",
       "        ['6.5', '2.8', '4.6', '1.5'],\n",
       "        ['5.7', '2.8', '4.5', '1.3'],\n",
       "        ['6.3', '3.3', '4.7', '1.6'],\n",
       "        ['4.9', '2.4', '3.3', '1'],\n",
       "        ['6.6', '2.9', '4.6', '1.3'],\n",
       "        ['5.2', '2.7', '3.9', '1.4'],\n",
       "        ['5', '2', '3.5', '1'],\n",
       "        ['5.9', '3', '4.2', '1.5'],\n",
       "        ['6', '2.2', '4', '1'],\n",
       "        ['6.1', '2.9', '4.7', '1.4'],\n",
       "        ['5.6', '2.9', '3.6', '1.3'],\n",
       "        ['6.7', '3.1', '4.4', '1.4'],\n",
       "        ['5.6', '3', '4.5', '1.5'],\n",
       "        ['5.8', '2.7', '4.1', '1'],\n",
       "        ['6.2', '2.2', '4.5', '1.5'],\n",
       "        ['5.6', '2.5', '3.9', '1.1'],\n",
       "        ['5.9', '3.2', '4.8', '1.8'],\n",
       "        ['6.1', '2.8', '4', '1.3'],\n",
       "        ['6.3', '2.5', '4.9', '1.5'],\n",
       "        ['6.1', '2.8', '4.7', '1.2'],\n",
       "        ['6.4', '2.9', '4.3', '1.3'],\n",
       "        ['6.6', '3', '4.4', '1.4'],\n",
       "        ['6.8', '2.8', '4.8', '1.4'],\n",
       "        ['6.7', '3', '5', '1.7'],\n",
       "        ['6', '2.9', '4.5', '1.5'],\n",
       "        ['5.7', '2.6', '3.5', '1'],\n",
       "        ['5.5', '2.4', '3.8', '1.1'],\n",
       "        ['5.5', '2.4', '3.7', '1'],\n",
       "        ['5.8', '2.7', '3.9', '1.2'],\n",
       "        ['6', '2.7', '5.1', '1.6'],\n",
       "        ['5.4', '3', '4.5', '1.5'],\n",
       "        ['6', '3.4', '4.5', '1.6'],\n",
       "        ['6.7', '3.1', '4.7', '1.5'],\n",
       "        ['6.3', '2.3', '4.4', '1.3'],\n",
       "        ['5.6', '3', '4.1', '1.3'],\n",
       "        ['5.5', '2.5', '4', '1.3'],\n",
       "        ['5.5', '2.6', '4.4', '1.2'],\n",
       "        ['6.1', '3', '4.6', '1.4'],\n",
       "        ['5.8', '2.6', '4', '1.2'],\n",
       "        ['5', '2.3', '3.3', '1'],\n",
       "        ['5.6', '2.7', '4.2', '1.3'],\n",
       "        ['5.7', '3', '4.2', '1.2'],\n",
       "        ['5.7', '2.9', '4.2', '1.3'],\n",
       "        ['6.2', '2.9', '4.3', '1.3'],\n",
       "        ['5.1', '2.5', '3', '1.1'],\n",
       "        ['5.7', '2.8', '4.1', '1.3'],\n",
       "        ['6.3', '3.3', '6', '2.5'],\n",
       "        ['5.8', '2.7', '5.1', '1.9'],\n",
       "        ['7.1', '3', '5.9', '2.1'],\n",
       "        ['6.3', '2.9', '5.6', '1.8'],\n",
       "        ['6.5', '3', '5.8', '2.2'],\n",
       "        ['7.6', '3', '6.6', '2.1'],\n",
       "        ['4.9', '2.5', '4.5', '1.7'],\n",
       "        ['7.3', '2.9', '6.3', '1.8'],\n",
       "        ['6.7', '2.5', '5.8', '1.8'],\n",
       "        ['7.2', '3.6', '6.1', '2.5'],\n",
       "        ['6.5', '3.2', '5.1', '2'],\n",
       "        ['6.4', '2.7', '5.3', '1.9'],\n",
       "        ['6.8', '3', '5.5', '2.1'],\n",
       "        ['5.7', '2.5', '5', '2'],\n",
       "        ['5.8', '2.8', '5.1', '2.4'],\n",
       "        ['6.4', '3.2', '5.3', '2.3'],\n",
       "        ['6.5', '3', '5.5', '1.8'],\n",
       "        ['7.7', '3.8', '6.7', '2.2'],\n",
       "        ['7.7', '2.6', '6.9', '2.3'],\n",
       "        ['6', '2.2', '5', '1.5'],\n",
       "        ['6.9', '3.2', '5.7', '2.3'],\n",
       "        ['5.6', '2.8', '4.9', '2'],\n",
       "        ['7.7', '2.8', '6.7', '2'],\n",
       "        ['6.3', '2.7', '4.9', '1.8'],\n",
       "        ['6.7', '3.3', '5.7', '2.1'],\n",
       "        ['7.2', '3.2', '6', '1.8'],\n",
       "        ['6.2', '2.8', '4.8', '1.8'],\n",
       "        ['6.1', '3', '4.9', '1.8'],\n",
       "        ['6.4', '2.8', '5.6', '2.1'],\n",
       "        ['7.2', '3', '5.8', '1.6'],\n",
       "        ['7.4', '2.8', '6.1', '1.9'],\n",
       "        ['7.9', '3.8', '6.4', '2'],\n",
       "        ['6.4', '2.8', '5.6', '2.2'],\n",
       "        ['6.3', '2.8', '5.1', '1.5'],\n",
       "        ['6.1', '2.6', '5.6', '1.4'],\n",
       "        ['7.7', '3', '6.1', '2.3'],\n",
       "        ['6.3', '3.4', '5.6', '2.4'],\n",
       "        ['6.4', '3.1', '5.5', '1.8'],\n",
       "        ['6', '3', '4.8', '1.8'],\n",
       "        ['6.9', '3.1', '5.4', '2.1'],\n",
       "        ['6.7', '3.1', '5.6', '2.4'],\n",
       "        ['6.9', '3.1', '5.1', '2.3'],\n",
       "        ['5.8', '2.7', '5.1', '1.9'],\n",
       "        ['6.8', '3.2', '5.9', '2.3'],\n",
       "        ['6.7', '3.3', '5.7', '2.5'],\n",
       "        ['6.7', '3', '5.2', '2.3'],\n",
       "        ['6.3', '2.5', '5', '1.9'],\n",
       "        ['6.5', '3', '5.2', '2'],\n",
       "        ['6.2', '3.4', '5.4', '2.3'],\n",
       "        ['5.9', '3', '5.1', '1.8']], dtype=object),\n",
       " array(['Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa',\n",
       "        'Setosa', 'Setosa', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Versicolor',\n",
       "        'Versicolor', 'Versicolor', 'Versicolor', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica', 'Virginica', 'Virginica',\n",
       "        'Virginica', 'Virginica', 'Virginica'], dtype=object))"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X = dataset[1:,0:4]#.astype(float)\n",
    "Y = dataset[1:,4]\n",
    "\n",
    "X, Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e583a9da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,\n",
       "       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,\n",
       "       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# encode class values as integers\n",
    "encoder = LabelEncoder()\n",
    "encoder.fit(Y)\n",
    "encoded_Y = encoder.transform(Y)\n",
    "encoded_Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "812269af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [1., 0., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 1., 0.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.],\n",
       "       [0., 0., 1.]], dtype=float32)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert integers to dummy variables (i.e. one hot encoded)\n",
    "dummy_y = np_utils.to_categorical(encoded_Y)\n",
    "dummy_y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c6f3e84c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define baseline model\n",
    "def baseline_model():\n",
    "    # create model\n",
    "    model = Sequential()\n",
    "    model.add(Dense(8, input_dim=4, activation='relu'))\n",
    "    model.add(Dense(3, activation='softmax'))\n",
    "    # Compile model\n",
    "    model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])\n",
    "    return model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b3ae175",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_8728/1754612998.py:1: DeprecationWarning: KerasClassifier is deprecated, use Sci-Keras (https://github.com/adriangb/scikeras) instead.\n",
      "  estimator = KerasClassifier(build_fn=baseline_model, epochs=200, batch_size=5, verbose=0)\n"
     ]
    }
   ],
   "source": [
    "estimator = KerasClassifier(build_fn=baseline_model, epochs=200, batch_size=5, verbose=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "57b0b4d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-23 10:59:14.952145: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory\n",
      "2021-12-23 10:59:14.952183: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)\n",
      "2021-12-23 10:59:14.952212: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (menny-Inspiron-5570): /proc/driver/nvidia/version does not exist\n",
      "2021-12-23 10:59:14.952427: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2021-12-23 10:59:15.350311: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:15.897981: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:16.252293: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:16.588200: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:16.908777: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:17.487307: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:17.872440: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:18.186657: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "2021-12-23 10:59:18.484175: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Baseline: nan% (nan%)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-23 10:59:18.806671: W tensorflow/core/framework/op_kernel.cc:1722] OP_REQUIRES failed at cast_op.cc:121 : UNIMPLEMENTED: Cast string to float is not supported\n",
      "/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:372: FitFailedWarning: \n",
      "10 fits failed out of a total of 10.\n",
      "The score on these train-test partitions for these parameters will be set to nan.\n",
      "If these failures are not expected, you can try to debug them by setting error_score='raise'.\n",
      "\n",
      "Below are more details about the failures:\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_615]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1043, in __call__\n",
      ">>>     if self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_1/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_1229]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_1/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_2/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_1843]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_2/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_3/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_2457]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_3/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_4/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_3071]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_4/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_5/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_3685]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_5/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_6/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_4299]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_6/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_7/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_4913]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_7/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_8/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_5527]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_8/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "--------------------------------------------------------------------------------\n",
      "1 fits failed with the following error:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      "    return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      "    history = self.model.fit(x, y, **fit_args)\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 67, in error_handler\n",
      "    raise e.with_traceback(filtered_tb) from None\n",
      "  File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tensorflow/python/eager/execute.py\", line 58, in quick_execute\n",
      "    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n",
      "tensorflow.python.framework.errors_impl.UnimplementedError:  Cast string to float is not supported\n",
      "\t [[node sequential_9/Cast\n",
      " (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py:671)\n",
      "]] [Op:__inference_train_function_6141]\n",
      "\n",
      "Errors may have originated from an input operation.\n",
      "Input Source operations connected to node sequential_9/Cast:\n",
      "In[0] IteratorGetNext (defined at /home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py:866)\n",
      "\n",
      "Operation defined at: (most recent call last)\n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      ">>>     return _run_code(code, main_globals, None,\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      ">>>     exec(code, run_globals)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel_launcher.py\", line 16, in <module>\n",
      ">>>     app.launch_new_instance()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/traitlets/config/application.py\", line 846, in launch_instance\n",
      ">>>     app.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 677, in start\n",
      ">>>     self.io_loop.start()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/tornado/platform/asyncio.py\", line 199, in start\n",
      ">>>     self.asyncio_loop.run_forever()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 570, in run_forever\n",
      ">>>     self._run_once()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/base_events.py\", line 1859, in _run_once\n",
      ">>>     handle._run()\n",
      ">>> \n",
      ">>>   File \"/usr/lib/python3.8/asyncio/events.py\", line 81, in _run\n",
      ">>>     self._context.run(self._callback, *self._args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 457, in dispatch_queue\n",
      ">>>     await self.process_one()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 446, in process_one\n",
      ">>>     await dispatch(*args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 353, in dispatch_shell\n",
      ">>>     await result\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/kernelbase.py\", line 648, in execute_request\n",
      ">>>     reply_content = await reply_content\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/ipkernel.py\", line 353, in do_execute\n",
      ">>>     res = shell.run_cell(code, store_history=store_history, silent=silent)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/ipykernel/zmqshell.py\", line 533, in run_cell\n",
      ">>>     return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2901, in run_cell\n",
      ">>>     result = self._run_cell(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 2947, in _run_cell\n",
      ">>>     return runner(coro)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      ">>>     coro.send(None)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3172, in run_cell_async\n",
      ">>>     has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3364, in run_ast_nodes\n",
      ">>>     if (await self.run_code(code, result,  async_=asy)):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py\", line 3444, in run_code\n",
      ">>>     exec(code_obj, self.user_global_ns, self.user_ns)\n",
      ">>> \n",
      ">>>   File \"/tmp/ipykernel_8728/3546967524.py\", line 2, in <module>\n",
      ">>>     results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 510, in cross_val_score\n",
      ">>>     cv_results = cross_validate(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 267, in cross_validate\n",
      ">>>     results = parallel(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 1046, in __call__\n",
      ">>>     while self.dispatch_one_batch(iterator):\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 861, in dispatch_one_batch\n",
      ">>>     self._dispatch(tasks)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 779, in _dispatch\n",
      ">>>     job = self._backend.apply_async(batch, callback=cb)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 208, in apply_async\n",
      ">>>     result = ImmediateResult(func)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/_parallel_backends.py\", line 572, in __init__\n",
      ">>>     self.results = batch()\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in __call__\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/joblib/parallel.py\", line 262, in <listcomp>\n",
      ">>>     return [func(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py\", line 211, in __call__\n",
      ">>>     return self.function(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 681, in _fit_and_score\n",
      ">>>     estimator.fit(X_train, y_train, **fit_params)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 232, in fit\n",
      ">>>     return super(KerasClassifier, self).fit(x, y, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/wrappers/scikit_learn.py\", line 164, in fit\n",
      ">>>     history = self.model.fit(x, y, **fit_args)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 1216, in fit\n",
      ">>>     tmp_logs = self.train_function(iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 878, in train_function\n",
      ">>>     return step_function(self, iterator)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 867, in step_function\n",
      ">>>     outputs = model.distribute_strategy.run(run_step, args=(data,))\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 860, in run_step\n",
      ">>>     outputs = model.train_step(data)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/training.py\", line 808, in train_step\n",
      ">>>     y_pred = self(x, training=True)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 64, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/base_layer.py\", line 1083, in __call__\n",
      ">>>     outputs = call_fn(inputs, *args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py\", line 92, in error_handler\n",
      ">>>     return fn(*args, **kwargs)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/sequential.py\", line 373, in call\n",
      ">>>     return super(Sequential, self).call(inputs, training=training, mask=mask)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 451, in call\n",
      ">>>     return self._run_internal_graph(\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 571, in _run_internal_graph\n",
      ">>>     y = self._conform_to_reference_input(y, ref_input=x)\n",
      ">>> \n",
      ">>>   File \"/home/menny/Documents/Progetti/prova1/venv/lib/python3.8/site-packages/keras/engine/functional.py\", line 671, in _conform_to_reference_input\n",
      ">>>     tensor = tf.cast(tensor, dtype=ref_input.dtype)\n",
      ">>> \n",
      "\n",
      "  warnings.warn(some_fits_failed_message, FitFailedWarning)\n"
     ]
    }
   ],
   "source": [
    "kfold = KFold(n_splits=10, shuffle=True)\n",
    "results = cross_val_score(estimator, X, dummy_y, cv=kfold)\n",
    "print(\"Baseline: %.2f%% (%.2f%%)\" % (results.mean()*100, results.std()*100))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b60fd23",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34a86d60",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
